javaagent-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/concurrent/ExecutorInstrumentationUtils.java

```diff
@@ -14,6 +14,43 @@
 /** Utils for concurrent instrumentations. */
 public class ExecutorInstrumentationUtils {
 
+  private static final ClassValue<Boolean> NOT_INSTRUMENTED_RUNNABLE_ENCLOSING_CLASS =
+      new ClassValue<Boolean>() {
+        @Override
+        protected Boolean computeValue(Class<?> enclosingClass) {
+          // Avoid context leak on jetty. Runnable submitted from SelectChannelEndPoint is used to
+          // process a new request which should not have context from them current request.
+          if (enclosingClass.getName().equals("org.eclipse.jetty.io.nio.SelectChannelEndPoint")) {
```

Comment:
Just an idea, but maybe those class names should be specified by instrumentation modules? Each instrumented library may bring its own set of excluded classes (also see `AdditionalLibraryIgnoresMatcher`) and maybe it'd better to define them in the lib instrumentation.

Reply:
Yeah! That's the approach Datadog has implemented, it would be nice for us to follow it at some point.

Reply:
https://github.com/DataDog/dd-trace-java/blob/master/dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/ExcludeFilterProvider.java

Reply:
Nice - I originally thought about adding some `void excludeClasses(ExcludeBuilder)` method to `InstrumentationModule`